Prevent clashing of function names in different modules

Uxio0 opened the pull request:
Fixes #15

Hi @peterbe,

First of all I would like to thank you for your awesome library.

I found some problem using it in a project not so small, that some modules can have same function name and it's very tedious to set `prefix` all the time.

What do you think of the solution I propose using `__module__` (I know the tests are not passing)? I think that would fix a lot of cases, even if some would still remain (for example, objects with the same function name or inherited in the same module)

Kind regards and thank you for your work

peterbe commented:
Try using `__qualname__` instead of name and see if that works. 

If I remember correctly, it was brought up here: https://github.com/peterbe/django-cache-memoize/issues/15 
But that the time, `django-cache-memoize` had to still support Python 2. Those days are gone.

Uxio0 commented:
> Try using `__qualname__` instead of name and see if that works.
> 
> If I remember correctly, it was brought up here: #15
> But that the time, `django-cache-memoize` had to still support Python 2. Those days are gone.

Nice, I haven't read that issue and I thought Python2.7 support was required. Even better.

I have changed that and fixed the tests.

peterbe commented:
Seems you just have a linting error. 
Oh, and you don't need to rebase your PRs.

Uxio0 commented:
Should be fixed now

peterbe commented:
Should be in 0.1.7 now https://pypi.org/project/django-cache-memoize/0.1.7/

Uxio0 commented:
Awesome, thanks @peterbe ! 👏
